pycode/memilio-epidata/memilio/epidata_test/test_epidata_modifyDataframeSeries.py

```diff
@@ -361,6 +384,19 @@ def test_insert_column_by_map(self):
         pd.testing.assert_frame_equal(df[old_cols], self.test_df1)
         pd.testing.assert_series_equal(df['inserted_col'], self.df_str_map_col)
 
+    def test_extract_subframe_based_on_dates_single_date(self):
+        # test if only dates from 2021-09-09 are returned
+        extracted_df = mDfS.extract_subframe_based_on_dates(
+            self.date_df, date(2021, 9, 9),
+            date(2021, 9, 9))
+        pd.testing.assert_frame_equal(self.test_date_df2, extracted_df)
+    
+    def test_extract_subframe_based_on_dates_multiple_dates(self):
+        # test if only dates from 2021-09-12 to 2021-09-14 are returned
+        extracted_df = mDfS.extract_subframe_based_on_dates(
+            self.date_df, date(2021, 9, 12),
+            date(2021, 9, 14))
+        pd.testing.assert_frame_equal(self.test_date_df1, extracted_df)
```

Comment:
Are there maybe some edge cases you could test? Else looks good!

Reply:
I added a test with moving average.